File: ios/Sasquatsh/Views/Profile/LegalLinksView.swift
import SwiftUI

struct LegalLinksView: View {
    var body: some View {
        VStack(spacing: 8) {
            legalLink("Terms of Service", url: "https://sasquatsh.com/terms", icon: "doc.text")
            legalLink("Privacy Policy", url: "https://sasquatsh.com/privacy", icon: "lock.shield")
            legalLink("Contact Us", url: "https://sasquatsh.com/contact", icon: "envelope")
        }
    }

    private func legalLink(_ title: String, url: String, icon: String) -> some View {
        Link(destination: URL(string: url)!) {
            HStack(spacing: 10) {
                Image(systemName: icon)
                    .foregroundStyle(Color.md3OnSurfaceVariant)
                    .frame(width: 20)
                Text(title)
                    .font(.md3BodyMedium)
                    .foregroundStyle(Color.md3OnSurface)
                Spacer()
                Image(systemName: "arrow.up.right")
                    .font(.system(size: 11))
                    .foregroundStyle(Color.md3OnSurfaceVariant)
            }
            .padding(.vertical, 6)
        }
    }
}

File: android/app/src/main/java/com/sasquatsh/app/views/profile/LegalLinksView.kt
package com.sasquatsh.app.views.profile

import androidx.compose.foundation.clickable
import androidx.compose.foundation.layout.*
import androidx.compose.material.icons.Icons
import androidx.compose.material.icons.automirrored.filled.OpenInNew
import androidx.compose.material.icons.outlined.Description
import androidx.compose.material.icons.outlined.Email
import androidx.compose.material.icons.outlined.Shield
import androidx.compose.material3.Icon
import androidx.compose.material3.MaterialTheme
import androidx.compose.material3.Text
import androidx.compose.runtime.Composable
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.graphics.vector.ImageVector
import androidx.compose.ui.platform.LocalUriHandler
import androidx.compose.ui.unit.dp

@Composable
fun LegalLinksView(
    modifier: Modifier = Modifier
) {
    val uriHandler = LocalUriHandler.current

    Column(
        modifier = modifier,
        verticalArrangement = Arrangement.spacedBy(8.dp)
    ) {
        LegalLinkRow(
            title = "Terms of Service",
            url = "https://sasquatsh.com/terms",
            icon = Icons.Outlined.Description,
            onClick = { uriHandler.openUri("https://sasquatsh.com/terms") }
        )
        LegalLinkRow(
            title = "Privacy Policy",
            url = "https://sasquatsh.com/privacy",
            icon = Icons.Outlined.Shield,
            onClick = { uriHandler.openUri("https://sasquatsh.com/privacy") }
        )
        LegalLinkRow(
            title = "Contact Us",
            url = "https://sasquatsh.com/contact",
            icon = Icons.Outlined.Email,
            onClick = { uriHandler.openUri("https://sasquatsh.com/contact") }
        )
    }
}

@Composable
private fun LegalLinkRow(
    title: String,
    url: String,
    icon: ImageVector,
    onClick: () -> Unit
) {
    Row(
        modifier = Modifier
            .fillMaxWidth()
            .clickable(onClick = onClick)
            .padding(vertical = 6.dp),
        verticalAlignment = Alignment.CenterVertically,
        horizontalArrangement = Arrangement.spacedBy(10.dp)
    ) {
        Icon(
            imageVector = icon,
            contentDescription = null,
            tint = MaterialTheme.colorScheme.onSurfaceVariant,
            modifier = Modifier.size(20.dp)
        )
        Text(
            text = title,
            style = MaterialTheme.typography.bodyMedium,
            color = MaterialTheme.colorScheme.onSurface,
            modifier = Modifier.weight(1f)
        )
        Icon(
            imageVector = Icons.AutoMirrored.Filled.OpenInNew,
            contentDescription = "Open link",
            tint = MaterialTheme.colorScheme.onSurfaceVariant,
            modifier = Modifier.size(14.dp)
        )
    }
}
